plc4j/integrations/opcua-server/src/main/java/org/apache/plc4x/java/opcuaserver/backend/Plc4xCommunication.java

```diff
@@ -195,93 +169,110 @@ public static NodeId getNodeId(String plcValue) {
     }
 
     public DataValue getValue(AttributeFilterContext.GetAttributeContext ctx, String tag, String connectionString) {
-        PlcConnection connection = null;
         DataValue resp = new DataValue(new Variant(null), StatusCode.BAD);
+        PlcConnection connection = null;
+        try {
 
-        //Check if we just polled the connection and it failed. Wait for the backoff counter to expire before we try again.
-        if (failedConnectionList.containsKey(connectionString)) {
-            if (System.currentTimeMillis() > failedConnectionList.get(connectionString) + DEFAULT_RETRY_BACKOFF) {
-                failedConnectionList.remove(connectionString);
-            } else {
-                logger.debug("Waiting for back off timer - " + ((failedConnectionList.get(connectionString) + DEFAULT_RETRY_BACKOFF) - System.currentTimeMillis()) + " ms left");
-                return resp;
+            //Check if we just polled the connection and it failed. Wait for the backoff counter to expire before we try again.
+            if (failedConnectionList.containsKey(connectionString)) {
+                if (System.currentTimeMillis() > failedConnectionList.get(connectionString) + DEFAULT_RETRY_BACKOFF) {
+                    failedConnectionList.remove(connectionString);
+                } else {
+                    logger.debug("Waiting for back off timer - " + ((failedConnectionList.get(connectionString) + DEFAULT_RETRY_BACKOFF) - System.currentTimeMillis()) + " ms left");
+                    return resp;
+                }
             }
-        }
 
-        //Try to connect to PLC
-        try {
-            connection = driverManager.getConnection(connectionString);            
-            logger.debug(connectionString + " Connected");
-        } catch (PlcConnectionException e) {
-            logger.error("Failed to connect to device, error raised - " + e);
-            failedConnectionList.put(connectionString, System.currentTimeMillis());
-            return resp;
-        }
-
-        if (!connection.getMetadata().canRead()) {
-            logger.error("This connection doesn't support reading.");
+            //Try to connect to PLC
             try {
```

Comment:
Question, as this is a PLC4x client that exposes things as OPCUA, wouldn't we want the client to be using the pool?